### DISCUSSION TITLE
Build process

### DISCUSSION THREAD

**Discussion by rboarman** [juftin/camply#48] [General]
Hi,

I am working on a couple pull requests for the project if that's okay.

While I am familiar with Python, I have not used setup.cfg in the past. 

What's your normal build and test cycle? What commands do you use?

I'm also curious how you run your tests?

Running "python setup.py install" does not seem to update the code. My changes are not being executed.

Thank you!

**Comment by juftin** [comment]
Hey @rboarman :

setup.cfg is simply a [declarative configuration file](https://setuptools.pypa.io/en/latest/userguide/declarative_config.html), setup.py still works as it should

I use [tox](https://github.com/juftin/camply/blob/main/tox.ini) to manage my tests - `pip install tox` + `tox` will run the tests

My automated build, test, deploy workflows are all managed by GitHub Actions in my [workflow files](https://github.com/juftin/camply/tree/main/.github/workflows)

**Comment by rboarman** [comment]
Thank you. So what do I run after I make a change?

Thanks!

**Reply by juftin** [reply]
If you have a forked repository, all workflows will run when a pull request is created against `main` 

To run the unit tests manually you can just type in `tox` into your command line (after installing `tox`) - tox spins up a virtual env, installs camply and its testing dependencies, and kicks off specified tests. New functionality will require new unit tests to be written
